File: c言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter2_プログラミングの力を養う/2.1からnまで足し合わせる。/addup.c
#include <stdio.h>

int addup1()
{
	int total = 0;
	for( int ii = 1; ii < 10; ii++ ) {
		total += ii;
	}

	return total;
}

int addup2(int num)
{
	int total = 0;
	for( int ii = 1; ii <= num; ii++ ) {
		total += ii;
	}

	return total;
}

int addup3(int num)
{
	int total = (1 + num) * num / 2;

	return total;
}

int main(void)
{
	int ret = addup1();
	printf("1から10までの加算結果：%d\n", ret);

	ret = addup2(10);
	printf("1からnまでの加算結果：%d\n", ret);

	ret = addup3(10);
	printf("工夫した加算結果：%d\n", ret);

	return 0;
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=c noexpandtab:

File: Java言語/Pythonで学ぶアルゴリズムの教科書 一生モノの知識と技術を身につける/chapter2_プログラミングの力を養う/2.1からnまで足し合わせる。/addup.java
class addup {
	static void addup1(){
		var total = 0;
		for( int ii = 0; 10 >= ii; ii++ ) {
			total += ii;
		}
		System.out.printf("合計点 %d\n", total);	// 合計点 55
	}

	static void addup2(int number){
		var total = 0;
		for( int ii = 0; number >= ii; ii++ ) {
			total += ii;
		}
		System.out.printf("合計点 %d\n", total);	// 合計点 55
	}

	static int addup3(int number){
		var total = (1 + number) * number / 2;

		return total;
	}

	public static void main(String[] args){
		addup1();

		addup2(10);

		var total = addup3(10);
		System.out.printf("合計点 %d\n", total);	// 合計点 55
	}
}

// vim: set ts=4 sts=4 sw=4 tw=0 ff=unix fenc=utf-8 ft=java noexpandtab:
